Fatras/include/ActsFatras/Kernel/detail/Interactor.hpp

```diff
@@ -96,14 +100,26 @@ struct Interactor {
     // decay check. needs to happen at every step, not just on surfaces.
     // TODO limit the stepsize when close to the lifetime limit to avoid
     //   overstepping and decaying the particle systematically too late
-    if (result.properTimeLimit < result.particle.properTime()) {
+    if ((result.properTimeLimit - result.particle.properTime() < result.properTimeLimit * properTimeTolerance) &&
+        result.isAlive) {
```

Comment:
I do not understand the `result.isAlive` check. It should never be true within the actor call since the propagation should get aborted as soon as it is. If you saw that this is needed than there is probably a bug somewhere and this just masks it.

Reply:
It's not a bug, it's a feature of the `Propagator` calling the action list one more time after the propagation terminates.

Reply:
I did not know that. In that case, I would place the is-alive check at the very beginning of the actor method, independent from the decay checks. There are other ways in which the particle could die and by checking it at the beginning we can assume that the particle is still alive for the rest of the function. Something like
```cpp
// actors are called once more after the propagation terminated
if (not result.isAlive) {
  return;
}
```

Reply:
Btw. you can remove the TODO above this code since this PR actually fixes it.

Reply:
Added fast exit and removed comment